### DISCUSSION TITLE
serverless-associate-waf threw TypeError: this.config.name.trim is not a function

### DISCUSSION THREAD

**Discussion by ky2ninh** [serverless/serverless#11401] [General]
When I ran serverless deploy, serverless-associate-waf threw an error:

Environment: darwin, node 18.9.0, framework 3.22.0, plugin 6.2.2, SDK 4.3.2
Credentials: Local, "default" profile
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
TypeError: this.config.name.trim is not a function
    at AssociateWafPlugin.updateWafAssociation (/Users/kninh/node_modules/serverless-associate-waf/lib/index.js:58:66)
    at PluginManager.runHooks (/usr/local/lib/node_modules/serverless/lib/classes/plugin-manager.js:530:15)
    at async PluginManager.invoke (/usr/local/lib/node_modules/serverless/lib/classes/plugin-manager.js:565:9)
    at async PluginManager.run (/usr/local/lib/node_modules/serverless/lib/classes/plugin-manager.js:604:7)
    at async Serverless.run (/usr/local/lib/node_modules/serverless/lib/serverless.js:174:5)
    at async /usr/local/lib/node_modules/serverless/scripts/serverless.js:771:9

Relevant portion of serverless.yml

provider: 
  name: aws
  runtime: nodejs16.x
  region: us-west-2

resources:
  Resources:
    WafAcl:
      Type: 'AWS::WAFv2::WebACL'      
      Properties:                
        Name: aota-public-oddc-waf
        Scope: REGIONAL
        DefaultAction: 
          Block: {}
        VisibilityConfig:
          CloudWatchMetricsEnabled: true
          MetricName: AotaPublicOddcWaf
          SampledRequestsEnabled: true
        Rules:
          - Name: AllowIPs
            Priority: 0
            Action:
              Allow: {}
            Statement: 
              IPSetReferenceStatement: 
                Arn: !GetAtt WafAllowListIpSet.Arn
            VisibilityConfig:
              CloudWatchMetricsEnabled: true
              MetricName: AotaPublicOddcAllowIPs
              SampledRequestsEnabled: true

custom:
  associateWaf:
    name: !Ref WafAcl 
    version: V2

plugins:
  - serverless-associate-waf

**Comment by ky2ninh** [comment]
Has anyone run into similar error and how did you fix it?

**Comment by pgrzesik** [comment]
Hey, the most likely answer here is that the plugin does not support Serverless v3. Looks like it wasn't updated for a long time and it seems like it depends on an undocumented internal API that was changed.
